core.coffee

```diff
@@ -434,6 +447,35 @@
 						url += '?' + queryOptions.join('&')
 					return url
 
+			compileBatch: (data) ->
+				if !utils.isArray(data)
+					throw new Error('Batch must be passed an array')
+				data = utils.cloneDeep(data)
```

Comment:
I was using `cloneDeep` because the `buildReq` function mutates the object it is passed, i will rewrite it to avoid that though as I think this is the best solution. Regarding `map` I agree we can use coffeescripts for loop as you explained to me the other day :). `omit` is present in underscore too but it should not be to hard to avoid it completely if we don't want to add the extra requirement